Clusters GA updates to the docs (#2642)
buildkite/docs: pages/agent/v3/unclustered_tokens.md

# Unclustered agent tokens

> > 🚧 This page documents a deprecated feature of Buildkite

Reviewer: ```suggestion
> 🚧 This page documents a deprecated Buildkite feature
```

> > Previously, agents only connected to Buildkite directly via a token which was created and managed by the processes described on this page. These tokens are now a deprecated feature of Buildkite, and are referred to as _unclustered agent tokens_. Unclustered agent tokens, however, are still available to customers who have not yet migrated their pipelines across to a [cluster](/docs/clusters/overview).

Reviewer: ```suggestion
> Previously, agents only connected to Buildkite via a token which was created and managed by the processes described on this page. These tokens are now a deprecated feature of Buildkite, and are referred to as _unclustered agent tokens_. Unclustered agent tokens, however, are still available to customers who have not yet migrated their pipelines to a [cluster](/docs/clusters/overview).
```

Author: Yeah, I wasn't keen on the placement of the word 'directly' here but here, I'm using this word to contrast how unclustered agent tokens connect to Buildkite with what we now call 'agent tokens' (formerly 'cluster tokens').

The newer 'agent tokens' connect to Buildkite via a cluster, whereas 'unclustered agent tokens' connect directly to Buildkite.

Therefore, I'll move the word 'directly' after 'connected'.

> > Agent tokens are now associated with clusters, such that these tokens connect to Buildkite through a specific cluster within an organization. Refer to the main [agent token](/docs/agent/v3/tokens) for details on how to create agent tokens for clusters.

Reviewer: ```suggestion
> _Agent tokens_ are now associated with clusters, such that these tokens connect to Buildkite through a specific cluster within an organization. Refer to the main [agent token](/docs/agent/v3/tokens) page for details on how to manage agent tokens for clusters.
```

Author: I adopted most of these great suggestions, thank you!

> Any new Buildkite organizations created after the official release of clusters on February 26, 2024 will not have the ability to create unclustered agents. Therefore, unclustered agent tokens will not be relevant to these organizations.

> Any Buildkite organizations created before February 26, 2024, have an _Unclustered_ area for managing _unclustered agents_ (accessible through _Agents_ > _Unclustered_ of the Buildkite interface), where an _unclustered agent_ refers to any agent that is not associated with a cluster.

Reviewer: ```suggestion
Any Buildkite organization created before February 26, 2024 has an _Unclustered_ area for managing _unclustered agents_ (accessible through _Agents_ > _Unclustered_ of the Buildkite interface), where an _unclustered agent_ refers to any agent that is not associated with a cluster.
```

> Any Buildkite organizations created before February 26, 2024, have an _Unclustered_ area for managing _unclustered agents_ (accessible through _Agents_ > _Unclustered_ of the Buildkite interface), where an _unclustered agent_ refers to any agent that is not associated with a cluster.

Reviewer: Have another look at this and the next two paragraphs (through the default token section). These are not super clear yet, but uff that's a lot mouthfuls of clustering and unclustering and organizations to get across!

I thiiiink (from reading this) that part of the challenge to describing this clearly is that people may have some unclustered agents and some "new world" agent tokens at the same time, as they slowly migrate?

Author: Thanks heaps for your review of this WIP so far! Great pickups on some typos there :grin:!
I'll have a look at the rest of these on Monday.

Author: Great to have a 2nd pair of eyes on this note, looking back at this now I can see how this is a tad confusing and I don't think I had all the details on the exact behaviour back then. Anyway, I will re-write to make this clearer, hopefully! 🤞

Author: Just for more info, customers with:

- any existing Buildkite orgs (i.e. created before the 26th of Feb) will have an 'unclustered' area in the 'Agent Clusters' page of the Buildkite interface
- Buildkite orgs created after the 26th of Feb will not have an 'unclustered' area in the 'Agent Clusters' page of the Buildkite interface. In other words, unclustered agents will not work with these newer Buildkite orgs.

Therefore, for existing customers who create Buildkite orgs after the 26th of Feb, it's likely that they may have some Buildkite orgs with an 'unclustered' area and others which don't.

If they only really manage a single Buildkite org, that it will comply with the rules explained above.

Your Buildkite organization's unclustered agent tokens area (accessible through _Agents_ > _Unclustered_ > _Agent Tokens_) may have the _Default agent registration token_, which is the original default token when your organization was created. If you had previously saved this token's value in a safe place, this token can be used for testing and development. However, it's recommended that you [create new, specific tokens](#creating-tokens) for each new environment.

## Using and storing tokens

An unclustered agent token is used by the Buildkite agent's [start](/docs/agent/v3/cli-start#starting-an-agent) command, and can be provided on the command line, set in the [configuration file](/docs/agent/v3/configuration), or provided using the [environment variable](/docs/pipelines/environment-variables) `BUILDKITE_AGENT_TOKEN`.

It's recommended you use your platform's secret storage (such as the [AWS Systems Manager Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-paramstore.html)) to allow for easier rollover and management of your agent tokens.

> ## Creating tokens

Reviewer: ```suggestion
## Create a token
```

For example:

```graphql
mutation {
  agentTokenCreate(input: {
    organizationID: "organization-id",
    description: "A description"
  }) {
    tokenValue
    agentTokenEdge {
      node {
        id
      }
    }
  }
}
```

> 📘 An unclustered agent token's value is only displayed once
> As soon as the unclustered agent token's value is displayed, copy its value and save it in a secure location.
> If you forget to do this, you'll need to create a new token to obtain its value.

You can find your `organization-id` in your Buildkite organization settings page, or by running the following GrapqQL query:

```graphql
query GetOrgID {
  organization(slug: "organization-slug") {
    id
  }
}
```

<!--alex ignore clearly-->

The token description should clearly identify the environment the token is intended to be used for (for example, `Read-only token for static site generator`), and is listed on the _Agent tokens_ page of the _Agents_ > _Unclustered_ area. (This page is accessible through _Agents_ > _Unclustered_ > _Agent Tokens_.)

It is possible to create multiple unclustered agent tokens using the GraphQL API.

> ## Revoking tokens

Reviewer: ```suggestion
## Revoke a token
```
